src/client/terminals/codeExecution/helper.ts

```diff
@@ -63,17 +72,43 @@ export class CodeExecutionHelper implements ICodeExecutionHelper {
                     normalizeOutput.resolve(normalized);
                 },
             });
-
+            // If there is no explicit selection, we are exeucting 'line' or 'block'.
+            if (activeEditor && activeEditor.selection && activeEditor!.selection.isEmpty) {
+                sendTelemetryEvent(EventName.EXECUTION_CODE, undefined, { scope: 'line' });
+            }
             // The normalization script expects a serialized JSON object, with the selection under the "code" key.
             // We're using a JSON object so that we don't have to worry about encoding, or escaping non-ASCII characters.
-            const input = JSON.stringify({ code });
+            const startLineVal = activeEditor && activeEditor.selection ? activeEditor.selection.start.line : 0;
```

Comment:
```suggestion
            const startLineVal = activeEditor?.selection?.start.line ?? 0;
```
You could make simliar change to below.